File: Solutions/0003-longest-substring-without-repeating-characters/solution.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l=0,r=0,ans=0;
        bool isPresent[256];
        memset(isPresent,false,sizeof(isPresent));
        while(r<s.length()) {
            if(isPresent[s[r]]) {
                while(s[l]!=s[r]) {
                    isPresent[s[l]]=false;
                    l++;
                }
                isPresent[s[l]]=false;
                l++;
            } else {
                isPresent[s[r]]=true;
                ans = max(ans , r-l+1);
                r++;
            }
        }
        return ans;
    }
};

File: Solutions/0003-longest-substring-without-repeating-characters/solution.go
func lengthOfLongestSubstring(s string) int {
    var start,end,ans int
    isExisting := make(map[byte]bool)
    len :=  len(s)
    for end < len {
        for isExisting[s[end]] == true {
            isExisting[s[start]] = false
            start++
        }
        ans = max(ans,end-start+1)
        isExisting[s[end]] = true
        end++
    }
    return ans
}
